### PR TITLE
🚨🚨🐛 Source Mailchimp: Multiple schema updates and fixes

### LINKED ISSUE #24559: Source Mailchimp: Add undeclared columns to spec
## What
In this [PR](https://github.com/airbytehq/airbyte/pull/23985), `BasicReadTest.test_read` test in the Connector Acceptance Tests was updated to fail if the connector produces
stream records which contain columns that haven't been declared in the spec.

Mailchimp currently fails the updated test. Its `acceptance-test-config.yml` was edited with the
`fail_on_extra_columns: false` parameter in [this PR](https://github.com/airbytehq/airbyte/pull/24552) in order to avoid this change from making the connector fail CAT.

We want to add the undeclared columns to the spec.

**How this will help**:
- Column selection is currently blocked for connectors that don't declare all columns that they pass. This
  is because turning on column selection will stop these undeclared columns from being sent, when they were
  previously being sent. Doing this allows us to enable column selection for Mailchimp!
- Users will better understand the shape of the data they'll receive, since the columns will match the spec.

## How
The following descriptions of streams that pass undeclared columns come from results of the failed connector acceptance test:

```
columns.txt:./source-mailchimp-2137-2023-03-26T03:18:19.9668958Z The lists stream has the following schema errors:
columns.txt:./source-mailchimp:2138:2023-03-26T03:18:19.9669501Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-2940-2023-03-26T03:18:20.0407144Z The campaigns stream has the following schema errors:
columns.txt:./source-mailchimp:2941:2023-03-26T03:18:20.0407831Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-7531-2023-03-26T03:18:20.2181045Z The email_activity stream has the following schema errors:
columns.txt:./source-mailchimp:7532:2023-03-26T03:18:20.2181337Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-7617-2023-03-26T03:18:20.2202964Z The reports stream has the following schema errors:
columns.txt:./source-mailchimp:7618:2023-03-26T03:18:20.2203250Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-8825-2023-03-26T03:18:20.3571177Z The lists stream has the following schema errors:
columns.txt:./source-mailchimp:8826:2023-03-26T03:18:20.3571475Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-9628-2023-03-26T03:18:20.3891627Z The campaigns stream has the following schema errors:
columns.txt:./source-mailchimp:9629:2023-03-26T03:18:20.3892156Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-14219-2023-03-26T03:18:20.5646132Z The email_activity stream has the following schema errors:
columns.txt:./source-mailchimp:14220:2023-03-26T03:18:20.5646559Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-14305-2023-03-26T03:18:20.5676807Z The reports stream has the following schema errors:
columns.txt:./source-mailchimp:14306:2023-03-26T03:18:20.5677316Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-15859-2023-03-26T03:19:58.3210335Z The lists stream has the following schema errors:
columns.txt:./source-mailchimp:15860:2023-03-26T03:19:58.3211325Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-16662-2023-03-26T03:19:58.3758088Z The campaigns stream has the following schema errors:
columns.txt:./source-mailchimp:16663:2023-03-26T03:19:58.3758610Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-21253-2023-03-26T03:19:58.6847157Z The email_activity stream has the following schema errors:
columns.txt:./source-mailchimp:21254:2023-03-26T03:19:58.6847597Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-21339-2023-03-26T03:19:58.6868845Z The reports stream has the following schema errors:
columns.txt:./source-mailchimp:21340:2023-03-26T03:19:58.6869129Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-22561-2023-03-26T03:19:58.9006934Z The lists stream has the following schema errors:
columns.txt:./source-mailchimp:22562:2023-03-26T03:19:58.9007481Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-23364-2023-03-26T03:19:58.9354196Z The campaigns stream has the following schema errors:
columns.txt:./source-mailchimp:23365:2023-03-26T03:19:58.9354479Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-27955-2023-03-26T03:19:59.1086346Z The email_activity stream has the following schema errors:
columns.txt:./source-mailchimp:27956:2023-03-26T03:19:59.1086632Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-28041-2023-03-26T03:19:59.1110597Z The reports stream has the following schema errors:
columns.txt:./source-mailchimp:28042:2023-03-26T03:19:59.1110891Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-29580-2023-03-26T03:21:37.1635687Z The lists stream has the following schema errors:
columns.txt:./source-mailchimp:29581:2023-03-26T03:21:37.1636603Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-30383-2023-03-26T03:21:37.2725956Z The campaigns stream has the following schema errors:
columns.txt:./source-mailchimp:30384:2023-03-26T03:21:37.2727058Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-34974-2023-03-26T03:21:37.5012198Z The email_activity stream has the following schema errors:
columns.txt:./source-mailchimp:34975:2023-03-26T03:21:37.5012702Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-35060-2023-03-26T03:21:37.5050203Z The reports stream has the following schema errors:
columns.txt:./source-mailchimp:35061:2023-03-26T03:21:37.5050852Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-36268-2023-03-26T03:21:37.6955188Z The lists stream has the following schema errors:
columns.txt:./source-mailchimp:36269:2023-03-26T03:21:37.6955713Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-37071-2023-03-26T03:21:37.7187633Z The campaigns stream has the following schema errors:
columns.txt:./source-mailchimp:37072:2023-03-26T03:21:37.7187920Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-41662-2023-03-26T03:21:37.8770454Z The email_activity stream has the following schema errors:
columns.txt:./source-mailchimp:41663:2023-03-26T03:21:37.8770744Z Additional properties are not allowed ('_links' was unexpected)
columns.txt:./source-mailchimp-41748-2023-03-26T03:21:37.8792207Z The reports stream has the following schema errors:
columns.txt:./source-mailchimp:41749:2023-03-26T03:21:37.8792497Z Additional properties are not allowed ('_links' was unexpected)

```

1. Add the missing properties indicated by the `Additional properties are not allowed ('<column>', 'column' were unexpected)` logs to the connector's spec.
2. Remove `fail_on_extra_columns: false` from the `acceptance-test-config.yml` file.
3. Commit changes to spec and `acceptance-test-config.yml` and open a PR.
4. Run tests on the connector, either automatically via CI or manually via the `/test` command
5. Profit!

Definition of done: Mailchimp passes CAT without declaring `fail_on_extra_columns: false`. If the
API starts sending over extra columns in the future, we will catch and fix them as part of the #connector-health
movement.

### PR DESCRIPTION
**Pull request by ChristoGrab**
## What

This PR introduces multiple breaking changes to Source Mailchimp's schema: 

Timestamp fields have been updated from `string` to airbyte-type `timestamp_with_timezone`. In order to accommodate this change, empty `string` values are now converted to `null` for incremental streams. This is necessary as there are multiple timestamp fields which, when they do not contain a value, will be returned as empty strings by the API, causing validation errors on the declared date-time format.

The `._links` field, which contains non-user relevant Mailchimp metadata, has been excluded from all requests, and removed from our schema (it was only declared in the Automations schema). This field was already being excluded from recently added streams, and since this update already constitutes a breaking change, now is a good time to filter it out entirely.

### Docs updates

- Updated Prerequisites and added API Key setup steps
- Added `HideInUI` tags to non-setup sections, and collapsible Troubleshooting section
- Merged sync modes/streams lists into unified table

### Misc

- Added suggested streams based on usage reports from Metabase.
- Moved logic to remove non-user relevant `._links` field from individual stream implementation to base MailchimpStream class `request_params`.
- Refactored individual stream implementations of nullifying empty string fields to base IncrementalMailchimpStream class
- Removed a duplicate stream_state object in `integration_tests/state.json`
- Updates to expected records

## 🚨🚨 User Impact
Due to the changes in type to all timestamp fields, and the removal of the `._links` fields from the schema, this is a breaking change affecting all Incremental streams. Customers will have to refresh their schemas and reset data in order to upgrade.

### CODE REVIEW

**Review by katmarkham — APPROVED** [review]

**Review by bnchrch — COMMENTED** [review]
Just one question

Does this change
> Updated credentials to required field in connector spec

break any existing syncs?

**Review by bnchrch — APPROVED** [review]

**Comment by ChristoGrab** [comment]
@tolik0 airbyte_type has been added to all date-time fields as requested 👍 

My understanding is specifying the "date-time" format in the schema doesn't change the underlying data type of these fields, but rather enforces the format of the string, in which case it shouldn't constitute a breaking change (I assumed this is why our acceptance tests do not fail on backwards compatibility when the format is added). Let me know if my understanding is incorrect though, my grasp of how destinations handle declared types is very limited.

**Review by tolik0 — APPROVED** [review]

File: airbyte-integrations/connectors/source-mailchimp/source_mailchimp/schemas/segments.json [hunk 1]
```diff
@@ -44,7 +46,7 @@
                 "type": ["null", "string"]
               },
               "value": {
-                "type": ["null", "string"]
+                "type": ["null", "integer", "number", "string"]
```
Comment (tolik0) [inline airbyte-integrations/connectors/source-mailchimp/source_mailchimp/schemas/segments.json:49]: Do we support multiple types in a single destination column? I believe that a destination column can only handle one type. Fields with multiple types need to be standardized via transformation.

**Comment by tolik0** [comment]
@ChristoGrab Can you please add type for the `/properties/recipients/properties/segment_opts/properties/conditions/items` field in the `automations` stream?

File: airbyte-integrations/connectors/source-mailchimp/source_mailchimp/schemas/segments.json (hunk 1 above)
Comment (ChristoGrab) [reply airbyte-integrations/connectors/source-mailchimp/source_mailchimp/schemas/segments.json]: Thanks for the clarification! I've updated this to be an additional undeclared property for now as our sandbox data is very limited, and have added the type for the nested `conditions/items` in the automations stream as requested.

Comment (ChristoGrab) [reply airbyte-integrations/connectors/source-mailchimp/source_mailchimp/schemas/segments.json]: Thanks for the clarification! I've updated this field to be an additional undeclared property for now as we don't have much sandbox data to test against, and I've also added the type for the array items in the automations stream as you requested.

File: airbyte-integrations/connectors/source-mailchimp/metadata.yaml [hunk 2]
```diff
@@ -22,7 +22,21 @@ data:
...
     oss:
       enabled: true
+  releases:
+    breakingChanges:
+      1.0.0:
+        message:
+          Version 1.0.0 introduces schema changes to all incremental streams.
+          A full schema refresh and data reset are required to upgrade to this version.
+          For more details, see our <a href='https://docs.airbyte.io/integrations/sources/mailchimp-migrations'>migration guide</a>.
+        upgradeDeadline: "2023-12-21"
```
Comment (katmarkham) [inline airbyte-integrations/connectors/source-mailchimp/metadata.yaml:32]: "2024-01-10"

**Review by katmarkham — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-mailchimp/metadata.yaml (hunk 2 above)
Comment (ChristoGrab) [reply airbyte-integrations/connectors/source-mailchimp/metadata.yaml]: Updated 👍

**Review by artem1205 — APPROVED** [review]

**Review by katmarkham — APPROVED** [review]

**Pull request merged**